src/js/cf-tables.js

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * cf-tables
+ * https://github.com/cfpb/cf-tables
+ *
+ * A public domain work of the Consumer Financial Protection Bureau
+ */
+
+( function( $ ) {
+  'use strict';
+
+  var SortableTable = function( table, options ) {
+    /*  At the moment, there are no default settings, but here's an object
+        for their future use! */
+    var defaults = {
+        },
+    // settings is defaults combined with user options
+    settings = {},
+    // rows is an Array of Arrays, serving as a model of the table
+    rows = [],
+    $table = $( table ),
+    $headercells = $( table ).find( '.sortable' );
+
+    /**
+     * Initializes the SortableTable
+     * @param { object } options - Customizble options object
+     */
+    function _init( options ) {
+      settings = $.extend( {}, defaults, options );
+      _clickHandler();
+      // If the following classes exist, start by sorting those columns.
+      $table.find( '.sortable__start-up, .sortable__start-up' ).click();
+    }
+
+    /**
+       * Sorting function for Array.sort()
+       *
+       * @param { number } sign - A number where a negative number indicates a
+       * reverse sort.
+       * @param { sortType } sortType - A string used for options such as integer sorting
+       * @returns function - A function to be used by the Array.sort method, where
+       * the parameters 'a' and 'b' is each an Array (of Arrays) to be sorted
+       */
+
+    function _arraySorter( sign, sortType ) {
+      return function( a, b ) {
+        // Set a and b to the first Array in each Array-of-Arrays
+        a = a[0];
+        b = b[0];
+
+        // For number sort, convert a & b to numbers.
+        if ( sortType === 'number' ) {
+          a = Number( a.replace(/[^\d.-]/g, '') );
+          b = Number( b.replace(/[^\d.-]/g, '') );
+        }
+
+        // Sort the values
+        if ( a < b ) {
+          return sign * -1;
+        }
+        if ( a > b ) {
+          return sign;
+        }
+        return 0;
+      };
+    }
+
+    /**
+     * Updates internal model of table (rows[])
+     * @param { index } - The index of the column used for sorting, which is
+     * used as the "key" for rows[] - it is set as the only value in the first
+     * array.
+     */
+    function _getRows( index ) {
+      // Clear the model
+      rows.length = 0;
+      // Find the value in each row of the column we're sorting by,
+      // add it to the rows Array
+      $table.find( 'tbody tr' ).each( function() {
+        // indices count from 0, but nth-child counts from 1
+        var child = index + 1,
+            key = $( this ).find( 'td:nth-child(' + child + ')' ).text().trim();
```

Comment:
Is there a reason to do so? I prefer variables to be declared as close to where they will be used as possible, and I don't think (because this variable is reset on every iteration) there's any performance difference.